src/ledgers/transactions.ts

```diff
@@ -264,13 +275,49 @@ export const claimRewards = async (
     })
   }
 
-  const fee = await getFee(address, ledgerType, [...msgAny], msgMemo)
+  fee = await getFee(address, ledgerType, [...msgAny], msgMemo)
 
   const client = await signingClient(ledgerType)
 
   const result = await client.signAndBroadcast(address, msgAny, fee, msgMemo)
 
-  return { result, fee: fee.amount[0].amount }
+  if (claimAndRestakeSeparateMsg) {
+    stakedValidators.forEach((validator) => {
+      if (Number(validator.amount) > 0) {
+        test.push({
+          typeUrl: '/cosmos.staking.v1beta1.MsgDelegate',
+          value: MsgUndelegate.fromPartial({
+            delegatorAddress: address,
+            validatorAddress: validator.address,
+            amount: {
+              amount: new BigNumber(
+                formatToken(validator.amount, 'cudos').value
+              )
+                .multipliedBy(CosmosNetworkConfig.CURRENCY_1_CUDO)
+                .toString(10),
+              denom: CosmosNetworkConfig.CURRENCY_DENOM
+            }
+          })
+        })
+      }
+    })
+
+    const client = await signingClient(ledgerType)
+
+    restakeFee = await getFee(address, ledgerType, [...msgAny], msgMemo)
+
+    restakeTx = await client.signAndBroadcast(address, test, fee, msgMemo)
+  }
+
+  if (restakeFee) {
+    fee = (
+      Number(fee.amount[0].amount) + Number(restakeFee.amount[0].amount)
+    ).toString()
+  } else {
```

Comment:
you can skip the else and use returns instead

Reply:
Done